concrete-core/docs/api/backends.md

```diff
@@ -0,0 +1,243 @@
+contains a blueprint (in the form of Rust Traits) of the FHE scheme exposed in `concrete-core`.
+- The `backends` module contains submodules (such a submodule, we call a _backend_ in this document), which implement all or a
+part of the specification.
+
+Those backends may rely on different hardware resources, which may not always be available. For this reason,
+it is possible to include and exclude backends from a build, by acting on the associated `backend_*` feature
+flag.
+
+## Entities and Engines
+
+The `specification` module describes two kinds of objects which can be implemented in a backend.
+
+All the traits appearing in the
+[`specification::entities`](https://docs.rs/concrete-core/latest/concrete_core/specification/entities/index.html)
+module represent the _datatypes_ manipulated in the library. To mention a few of them, we have:
++ [`PlaintextEntity`](https://docs.rs/concrete-core/latest/concrete_core/specification/entities/trait.PlaintextEntity.html)
++ [`LweSecretKeyEntity`](https://docs.rs/concrete-core/latest/concrete_core/specification/entities/trait.LweSecretKeyEntity.html)
++ [`LweCipertextEntity`](https://docs.rs/concrete-core/latest/concrete_core/specification/entities/trait.LweCiphertextEntity.html)
++ and more
+
+Only _one_ of the `*Entity` traits can be implemented at once, by a type exported by a backend. If a structure
+implements `PlaintextEntity`, it can not be `LweCiphertextEntity` at the same time, for instance.
+
+All the traits appearing in the
+[`specification::engines`](https://docs.rs/concrete-core/latest/concrete_core/specification/engines/index.html)
+module represent the _operators_ which can be used to manipulate entities in the library. For instance, we
+have:
++ [`LweSecretKeyCreationEngine`](https://docs.rs/concrete-core/latest/concrete_core/specification/engines/trait.LweSecretKeyCreationEngine.html)
++ [`LweCiphertextEncryptionEngine`](https://docs.rs/concrete-core/latest/concrete_core/specification/engines/trait.LweCiphertextEncryptionEngine.html)
++ [`LweCiphertextDecryptionEngine`](https://docs.rs/concrete-core/latest/concrete_core/specification/engines/trait.LweCiphertextDescryptionEngine.html)
++ and more
+
+If you read between the lines, the fact that we use traits to represent operators, means that we will have to
+use special object (which we call _engines_ in this document), to perform the operations. This is
+slightly different from an object model in which operators are usually tied to the data themselves. In
+practice, this makes sense because we tend to use _side-resources_ to perform operations, and those are better
+stored in a separate structure.
+
+Contrary to entities, there is no restrictions on how many `*Engine` traits a type can implement. In practice,
+we tend to expose very few engines in a given backend.
+
+## Example backend
+
+Let's go through an example of a backend implementation to ensure we understand the concepts clearly.
```

Comment:
"to ensure we understand the concepts clearly" maybe "to illustrate the concepts"?